zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -329,6 +335,16 @@ private Stream<String> getAuthorizedResourceIdentifiers(
     return Stream.concat(direct, Stream.concat(viaRole, viaGroups));
   }
 
+  private List<String> fetchGroups(
+      final TypedRecord<?> command, final EntityType ownerType, final String ownerId) {
+    final List<String> groupsClaims =
+        (List<String>) command.getAuthorizations().get(Authorization.GROUPS_CLAIMS);
+    if (groupsClaims != null) {
+      return groupsClaims;
+    }
```

Comment:
💭 So if a customer decides to bring their own groups, our group management is not usable at all? I wonder if we should somehow catch that and prevent users from making their own groups in the first place.

👍 I like the return early approach, no need to query RocksDB.